### PR TITLE
EVG-3099 also check for old task ID in evergreen contexts

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: model/context.go [hunk 1]
```diff
@@ -89,7 +89,14 @@ func (ctx *Context) populateTaskBuildVersion(taskId, buildId, versionId string)
...
 		ctx.Task, err = task.FindOne(task.ById(taskId))
 		if err != nil {
-			return "", err
+			// if no task found, see if this is an old task
+			tasks, err := task.FindOld(task.ById(taskId))
+			if err != nil {
+				return "", err
+			}
+			if len(tasks) == 1 {
+				ctx.Task = &tasks[0]
```
Comment (richardsamuels) [inline model/context.go]: What are the other tasks that can come up? Other executions?

Comment (john-m-liu) [reply model/context.go]: It shouldn't be possible for any other tasks to come up - this is a sanity check

Comment (richardsamuels) [reply model/context.go]: Maybe through an ErrorWhen(len(tasks) > 1)? otherwise lgtm

Comment (tychoish) [reply model/context.go]: I think this is fine because it's an id query. It's also legit to add a task.FindOldOne() helper to the packag.e

**Pull request merged**